### ISSUE TITLE
Runtime error "Method override...cannot find a method to replace"

### ISSUE THREAD

**Issue by patluze** [yck1509/ConfuserEx#220]
I ran ConfuserEx on a library DLL (MyAssembly.dll) using only rename protection.  ConfuserEx ran with no errors but when I ran an application that used the confused DLL, I received an error message:

Method override '=rnfwpc4GuNQGgdUkrGDWKc6TG5=' on type '=BRTrRgyPlry3oCmCELLI0fSqoc=' from assembly 'MyAssembly, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' cannot find a method to replace.

The Stack Trace Decoder translated the message to:

Method override ' Method1' on type 'C' from assembly 'MyAssembly, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' cannot find a method to replace.

Method1 is defined in class C of a hierarchy that looks like this:

internal class D : C\<Interface5, Interface6>
{
…

internal abstract class C\<T, U> : B\<U>, Interface1, Interface2, Interface3, Interface4
{
protected internal override bool Method1(object obj)
{
…

internal abstract class B\<T> : A
{
protected internal abstract bool Method1(object obj);
 …

When I look at the confused IL I see that the class D type has not been obfuscated despite being an internal class. The A, B, and C types have been obfuscated.  My questions are:

1)	Class D isn’t a public class – why isn't it obfuscated?
2)	The best I could do to avoid the runtime error was to specify some rules that keep all the classes in the hierarchy unobfuscated, but my goal is to obfuscate as much as possible except the public types. I’m looking for options that get me closer to my goal.

Thanks in advance for any help, and for the work you put into ConfuserEx.

**Comment by yck1509** [comment]
Hi,
Sorry I can't reproduce the issue according to your code. Do you mind including a complete sample that demonstrate the problem?

**Comment by patluze** [comment]
Thanks for taking a look at this.  

While putting together a sample that demonstrates the problem, I found the answer to question 1) from my original post -- ConfuserEx wasn't obfuscating class D because I use it in a typeof statement from a different class in the DLL. I believe this is by design.

Coincidentally, the typeof statement appears to be the cause of the runtime error.  I've created a Visual Studio 2013 solution (on a Windows 7 64-bit system) that demonstrates the error.  The solution contains a C# library project and a C# console application project that references the library.  I confuse the DLL but not the application.  The error occurs if I specify .NET 2.0 as the target framework, but not if I specify .NET 4.5.

I can send or upload the entire solution (tell me how/where) or just include the source code in a comment here.  Let me know what will make it easiest for you.
